include/tst_test.h

```diff
@@ -23,6 +23,9 @@
 #endif /* __TEST_H__ */
 
 #include <unistd.h>
+#ifndef __GLIBC__
+#include <limits.h>
+#endif
 
```

Comment:
yeah, anytime you start writing tests around C-library specific macros (like `__GLIBC__`), you need to take a step back and figure out if that's really the best route.  chances are pretty good it is not ;).